### PR TITLE
Support SMP for AArch64

### PR DESCRIPTION
**Pull request by equation314**
Works fine on both qemu and the real machine.

### CODE REVIEW

File: kernel/src/arch/aarch64/board/raspi3/timer.rs [hunk 1]
```diff
@@ -8,12 +9,17 @@ pub fn init() {
     info!("timer: init end");
 }
 
-/// Returns the current time in microseconds.
+/// Set next timer interrupt to 10 ms from now.
```
Comment (wangrunji0408) [inline kernel/src/arch/aarch64/board/raspi3/timer.rs]: Mistake?

File: kernel/src/arch/aarch64/memory.rs [hunk 2]
```diff
@@ -4,8 +4,13 @@ use super::paging::MMIOType;
 use crate::consts::{KERNEL_OFFSET, MEMORY_OFFSET};
 use crate::memory::{init_heap, kernel_offset, Linear, MemoryAttr, MemorySet, FRAME_ALLOCATOR};
 use log::*;
+use spin::Mutex;
 use rcore_memory::PAGE_SIZE;
 
+lazy_static! {
+    static ref KERNEL_MEMORY_SET: Mutex<Option<MemorySet>> = Mutex::new(None);
```
Comment (wangrunji0408) [inline kernel/src/arch/aarch64/memory.rs]: `lazy_static` is unnecessary?

File: kernel/src/arch/aarch64/board/raspi3/mod.rs [hunk 3]
```diff
@@ -10,21 +10,24 @@ pub mod timer;
 
 use crate::drivers::gpu::fb::{self, ColorDepth, ColorFormat, FramebufferInfo, FramebufferResult};
 
+pub const BOARD_NAME: &'static str = "Raspberry Pi 3";
 pub const PERIPHERALS_START: usize = bcm2837::addr::PERIPHERALS_START;
 pub const PERIPHERALS_END: usize = bcm2837::addr::PERIPHERALS_END;
+pub const CPU_NUM: usize = 4;
+
+#[no_mangle]
+pub static CPU_SPIN_TABLE: [usize; CPU_NUM] = [0xd8, 0xe0, 0xe8, 0xf0];
```
Comment (wangrunji0408) [inline kernel/src/arch/aarch64/board/raspi3/mod.rs]: What's the meaning of these magic numbers?

File: kernel/src/arch/aarch64/board/raspi3/timer.rs (hunk 1 above)
Comment (equation314) [reply kernel/src/arch/aarch64/board/raspi3/timer.rs]: fixed

File: kernel/src/arch/aarch64/memory.rs (hunk 2 above)
Comment (equation314) [reply kernel/src/arch/aarch64/memory.rs]: fixed

File: kernel/src/arch/aarch64/board/raspi3/mod.rs (hunk 3 above)
Comment (equation314) [reply kernel/src/arch/aarch64/board/raspi3/mod.rs]: See https://github.com/torvalds/linux/blob/master/arch/arm/boot/dts/bcm2837.dtsi#L46 and https://github.com/qemu/qemu/blob/master/hw/arm/raspi.c#L67. Secondary CPUs will spin until the spin table holds a jump address.

Comment (wangrunji0408) [reply kernel/src/arch/aarch64/board/raspi3/mod.rs]: Got it

**Pull request merged**